multivariate_single_forecast.py

```diff
@@ -1,128 +1,119 @@
-                                           input_shape=x_train_single.shape[-2:]))
-# single_step_model.add(tf.keras.layers.LSTM(32, input_shape=x_train_single.shape[-2:]))
-
-single_step_model.add(tf.keras.layers.LSTM(16, activation='relu'))
-if CLASSIFICATION:
-  single_step_model.add(tf.keras.layers.Dense(16, activation='relu'))
-  single_step_model.add(tf.keras.layers.Dense(1, activation='sigmoid'))
-else:
-  single_step_model.add(tf.keras.layers.Dense(1))
-
-# single_step_model.compile(optimizer=tf.keras.optimizers.RMSprop(), loss='mae')
-single_step_model.compile(optimizer=tf.keras.optimizers.Adam(), loss=tf.keras.losses.BinaryCrossentropy(), metrics=['accuracy'])
-
-val_callback = tf.keras.callbacks.ModelCheckpoint(
-    'checkpoints/multivariate_single_model', monitor='val_accuracy', verbose=1, save_best_only=True,
-    save_weights_only=False, mode='auto', save_freq='epoch'
-)
-
-single_step_history = single_step_model.fit(train_data_single, epochs=EPOCHS,
-                                            steps_per_epoch=EVALUATION_INTERVAL,
-                                            validation_data=val_data_single,
-                                            validation_steps=50, callbacks=[val_callback])
-
-
-plot_train_history(single_step_history,
-                   'Single Step Training and validation loss')
-
-
-for x, y in val_data_single.take(10):
-  y_pred = single_step_model.predict(x)[0]
-  print(f"prediction: {y_pred}")
-  if CLASSIFICATION:
-    if y_pred >= 0.5:
-      y_pred = 1
-    else:
-      y_pred = 0
-  plot = show_plot([x[0][:, 0].numpy(), y[0].numpy(),
-                    y_pred], target_distance,
-                   'Single Step Prediction')
-  plot.show()
+def get_lstm():
+  """
+  Keras LSTM Architecture 
+  """
+  ssm = tf.keras.models.Sequential()
+
+  ssm.add(tf.keras.layers.LSTM(64, return_sequences=True,
+                                input_shape=xt.shape[-2:]))
+
+  ssm.add(tf.keras.layers.LSTM(32, return_sequences=True))
+
+  ssm.add(tf.keras.layers.LSTM(16))
+
+  ssm.add(tf.keras.layers.Dense(8))
+
+  ssm.add(tf.keras.layers.Dense(1, activation='sigmoid'))
+
+  ssm.compile(optimizer=tf.keras.optimizers.Adam(lr=0.1), 
+              loss=tf.keras.losses.BinaryCrossentropy(), 
+              metrics=['accuracy'])
+  return ssm
+
+if __name__ == "__main__":
+  # construct data and get additional info 
+  e = EquityData(DATA_DIR, DATA_SYM)
+  data = pyfinancialdata.get_multi_year(provider='histdata', instrument='SPXUSD', years=[2016, 2017, 2018], time_group='10min')
+  e.data = data
+
+  # add MACD
+  e.data['MA_long'] = e.data['close'].rolling(window=52).mean()
+  e.data['MA_short'] = e.data['close'].rolling(window=7).mean()
+
+  # evaluation interval
+  window = int(e.data.shape[0]/BATCH_SIZE) * 1
+  
+  # pick selected features
+  features = e.data[FEATURES]
+  
+  # cleanse
+  features.index = e.data.index
+  features = features.dropna()
+  features = features[26:]
```

Comment:
I guess this line is not needed, this was meant to drop NA because of moving average, since we use MA_long = 52 then we would need to drop the first 52, but I think `dropna()` above takes care of this. 

Also, 52 MA kind of big, we loose 52 days of data, maybe not worth it.

Reply:
ill let that decision be for a later chat